src/Compilers/Core/Portable/FileSystem/RelativePathResolver.cs

```diff
@@ -35,9 +35,9 @@ public RelativePathResolver(ImmutableArray<string> searchPaths, string baseDirec
             BaseDirectory = baseDirectory;
         }
 
-        public string ResolvePath(string reference, string baseFilePath)
+        internal string ResolvePath(IFileSystem fileSystem, string reference, string baseFilePath)
```

Comment:
>IFileSystem fileSystem [](start = 36, length = 22)

I'd prefer we don't tie this class with `IFileSystem`. We'll need to make the resolver public at some point. It'd be better to define a subclass that uses IFileSystem.

Reply:
I'd prefer to not have to sub-class a half dozen types in the compilerc code base every time I want to write a unit test that doesn't depend on the file system. This change is a reaction to attempting to use the current abstractions, throwing up my hands in frustration and pushing for a simpler more re-usable approach.

Reply:
Ok, let's define a subclass of the RelativePathResolver then that uses IFileSystem.

Reply:
Right and now I have to sub-class `CommonCompiler` to create instances of that class instead of that sub class. Once you force a subclass anywhere you end up having to subclass through the system.

Reply:
You don't need to subclass CommonCompiler. You just need to pass the subclass of RelativePathResolver to LoggingMetadataFileReferenceResolver.

Reply:
In CommonCompiler.GetCommandLineMetadataReferenceResolver you'd create an instance of the subclass instead of RelativePathResolver and pass it IFileSystem .

Reply:
Let me try this approach

Reply:
Okay this is working out rather well and reduces the churn here by a lot.